### DISCUSSION TITLE
ICU package issues on Azure Linux 3.0

### DISCUSSION THREAD

**Discussion by ivor-microsoft** [dotnet/dotnet-docker#6057] [General]
Hello, 

I am using the Azure Linux 3.0 Image mcr.microsoft.com/dotnet/aspnet:8.0-azurelinux3.0-amd64@sha256:792497953e1d624f353bcaa5da45dfabbac692e0945fca1786bb6f8522b1f6c6
and I am trying to start a process that currently crashes with the following error:
Process terminated. Couldn't find a valid ICU package installed on the system. Please install libicu (or icu-libs) using your package manager and try again. Alternatively you can set the configuration flag System.Globalization.Invariant to true if you want to run with no globalization support. Please see https://aka.ms/dotnet-missing-libicu for more information.
After this message I tried:

- ENV DOTNET_SYSTEM_GLOBALIZATION_INVARIANT=0
- tdnf install -y icu icu-devel
- But I got exactly the same issue

Just to make sure I turned on Invariant Localization 

- ENV DOTNET_SYSTEM_GLOBALIZATION_INVARIANT=1
That actually fixed the issue for this process but unfortunately I have another process that requires Globalization and then this other process crashes.

Can anyone, give me a help in how to fix this?
 
Thanks.

**Comment by lbussell** [comment]
Hey @ivor-microsoft, first please try the latest image. I was unable to reproduce this using the latest digest `mcr.microsoft.com/dotnet/aspnet:8.0-azurelinux3.0-amd64@sha256:e72e750b0f4b79b80b4ffa6bc68beb8aed4251c45c33e5976c5a2cf9beaa00c0`.

Second, which globalization related functionality are you trying to use that's causing the issue?
For example, I tried some simple currency formatting with this Dockerfile:

```Dockerfile
FROM mcr.microsoft.com/dotnet/sdk:8.0-azurelinux3.0 AS sdk
RUN dotnet new console -n App -o /src --no-restore
WORKDIR /src

COPY <<EOF /src/Program.cs
using System.Globalization;
using static System.Console;
string environmentVariable = Environment.GetEnvironmentVariable("DOTNET_SYSTEM_GLOBALIZATION_INVARIANT") ?? "not defined";

WriteLine();
WriteLine($"DOTNET_SYSTEM_GLOBALIZATION_INVARIANT: {environmentVariable}");
WriteLine();
WriteLine("Testing some globalization functionality:");

const int Value = 1337;
WriteLine($"Value: {Value.ToString()}");
WriteLine($"  en-US currency: {Value.ToString("c", new CultureInfo("en-US"))}");
WriteLine($"  jp-JP currency: {Value.ToString("c", new CultureInfo("jp-JP"))}");
WriteLine();
EOF

RUN dotnet restore
RUN dotnet publish --no-restore -o /app

FROM mcr.microsoft.com/dotnet/aspnet:8.0-azurelinux3.0 AS runtime
COPY --from=sdk /app /app/
ENTRYPOINT ["/app/App"]
```

And my output was:

```console
PS> docker build --pull --no-cache -t test -f .\Dockerfile .
PS> docker run --rm test

DOTNET_SYSTEM_GLOBALIZATION_INVARIANT: not defined

Testing some globalization functionality:
Value: 1337
  en-US currency: $1,337.00
  jp-JP currency: ¥ 1337
```

If you have a repro like this that would really help here. Thanks!
